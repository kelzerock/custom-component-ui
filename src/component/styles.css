html,
body {
  height: 100%;
  margin: 0;
}

.wrapper {
  margin: 0 auto;
  max-width: 800px;
}
