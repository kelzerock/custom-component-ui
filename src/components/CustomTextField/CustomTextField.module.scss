:root {
  --color-main-blue: #1976d2;
  --color-main-blue-100: #2070c0;
  --color-white: #fff;
  --color-white-100: #f6fafd;
  --color-red: #d32f2f;
  --color-gray: #f0f0f0;
  --color-dark-gray: #00000099;
}

@mixin shrinkLabel {
  top: 0px;
  left: 10px;
  padding: 1px 3px;
  font-size: 1.2rem;
  border-radius: 3px;
  transform: translateY(-50%);
}

.wrapper {
  position: relative;
  margin: 10px;
}


.label {
  background-color: var(--color-white);
  color: var(--color-dark-gray);
  font-size: 1.7rem;
  font-weight: normal;
  position: absolute;
  pointer-events: none;
  left: 10px;
  top: 50%;
  transform: translateY(-50%);
  transition: top 300ms ease, font-size 300ms ease;

  &.shrink {
    @include shrinkLabel();
  }
}

.error .label {
  border-color: var(--color-red);
  color: var(--color-red);
}

.input {
  min-height: 30px;
  min-width: 50px;
  border: solid 2px var(--color-gray);
  background: none;
  background-color: white;
  font-size: 18px;
  padding: 10px 10px 10px 5px;
  display: block;
  border-radius: 5px;
}

.error .input {
  border: solid 2px var(--color-red);
}

.input:focus {
  outline: none;
  border-color: var(--color-main-blue);
}

.error .input:focus {
  outline: none;
  border-color: var(--color-red);
}


.input:focus~.label {
  @include shrinkLabel();
  color: var(--color-main-blue)
}

.error .input:focus~.label {
  @include shrinkLabel();
  color: var(--color-red)
}

.shrink~.label {
  @include shrinkLabel();
}