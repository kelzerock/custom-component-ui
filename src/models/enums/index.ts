export { Checkbox } from "./enums-for-test/Checkbox";
export { Switch } from "./enums-for-test/Switch";
export { Modal } from "./enums-for-test/Modal";
