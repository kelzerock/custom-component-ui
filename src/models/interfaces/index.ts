export { CustomSelectProps } from "./custom-select-props";
export { CustomButtonProps } from "./custom-button-props";
export { CustomTextFieldProps } from "./custom-text-field-props";
export { OptionsData } from "./option-data";
