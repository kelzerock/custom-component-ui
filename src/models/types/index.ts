export { ButtonSize } from "./button-size";
export { ButtonVariant } from "./button-variant";
