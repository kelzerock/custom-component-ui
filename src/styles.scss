:root {
  --color-main-blue: #1976d2;
  --color-main-blue-100: #2070c0;
  --color-white: #fff;
  --color-white-100: #f6fafd;
  --color-red: #d32f2f;
  --color-gray: #f0f0f0;
  --color-dark-gray: #00000099;
}