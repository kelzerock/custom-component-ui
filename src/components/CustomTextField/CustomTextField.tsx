import { FC, useState } from "react";
import styles from "./CustomTextField.module.scss";
import { CustomTextFieldProps } from "../../models/interfaces";
import clsx from "clsx";
import { TextField } from "../../models/enums";

export const CustomTextField: FC<CustomTextFieldProps> = ({
  label,
  error,
  defaultValue,
  value,
  className,
  onChange,
  ...props
}: CustomTextFieldProps) => {
  const [valueInput, setValueInput] = useState(defaultValue ?? "");

  const handleChange = (event: React.FormEvent<HTMLInputElement>) => {
    const { target } = event;
    if (target instanceof HTMLInputElement) {
      const { value } = target;
      setValueInput(value);
    }
  };

  return (
    <div
      className={clsx(styles.wrapper, error && styles.error)}
      data-testid={TextField.wrapper}
    >
      <input
        type="text"
        className={clsx(
          styles.input,
          (value || (typeof valueInput === "string" && valueInput.trim())) &&
            styles.shrink,
          className,
        )}
        value={value ?? valueInput}
        onChange={onChange ?? handleChange}
        {...props}
        data-testid={TextField.input}
      ></input>
      {label && (
        <label className={styles.label} data-testid={TextField.label}>
          {label}
        </label>
      )}
    </div>
  );
};
