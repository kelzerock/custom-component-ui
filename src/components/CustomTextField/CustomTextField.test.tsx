import { render, screen } from "@testing-library/react";
import { CustomTextField } from "./CustomTextField";
import { TextField } from "../../models/enums";
import styles from "./CustomTextField.module.scss";

describe("check CustomTextField", () => {
  test("check input element", () => {
    const value = "test value";
    render(<CustomTextField value={value} />);

    const textField = screen.getByTestId(TextField.input);
    expect(textField).toHaveAttribute("type", "text");
    expect(textField).toHaveAttribute("value", value);
    expect(textField).toHaveClass(styles.input);
  });

  test("check custom 'error' props", () => {
    render(<CustomTextField error={true} />);
    const wrapper = screen.getByTestId(TextField.wrapper);

    expect(wrapper).toHaveClass("error");
  });

  test("check UI snapshot", () => {
    const { container } = render(<CustomTextField label="test" error />);
    expect(container).toMatchSnapshot();
  });
});
