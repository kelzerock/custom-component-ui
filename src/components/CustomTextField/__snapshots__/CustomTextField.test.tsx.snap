// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check CustomTextField check UI snapshot 1`] = `
<div>
  <div
    class="wrapper error"
    data-testid="wrapper"
  >
    <input
      class="input "
      type="text"
    />
    <label
      class="label"
    >
      test
    </label>
  </div>
</div>
`;
